source/usage-examples/insertOne.txt

```diff
@@ -3,3 +3,52 @@ Insert a Document
 =================
 
 .. default-domain:: mongodb
+
+You can insert a document into a collection by using the ``InsertOne()``
+method.
+
+The following example creates a new document containing a ``title`` and ``text``
+field and inserts it into the ``haikus`` collection of the ``insertDB`` database:
+
+.. note:: Non-existent Database and Collections
+
+    The driver automatically creates the necessary database and/or collection
+    when you perform a write operation against them, if they don't exist.
+
+
+.. include:: /includes/usage-examples/run-example-tip.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/insertOne.go
+   :start-after: begin insertOne
+   :end-before: end insertOne
+   :emphasize-lines: 4
+   :language: go
+   :dedent:
+
+Click `here <{+example+}/insertOne.go>`__ to see a fully runnable example.
+
+Expected Result
+---------------
+
+After you run the code example, a new document with an ``_id`` that matches
+the output will be present:
```

Comment:
Suggestion: Something about this wording is awkward. It's unclear where the matching document is "present"." I also don't think the other usage examples have really been referencing the output of the full code. Maybe:

```suggestion
After you run the code example, you should be able to find the following document in the ``haikus`` collection.
```